src/mqtt.client.ts

```diff
@@ -123,22 +121,39 @@ export class MqttClient<
             );
     }
 
-    public async connect(options?: Resolvable<RegisterClientOptions>): Promise<any> {
+    private async _connect(options?: Resolvable<RegisterClientOptions>): Promise<any> {
         this.expectCreated();
-        this.mqttDebug('Connecting...');
+        this.mqttDebug(`Connecting using transport "${this.transport.constructor.name}"`);
         this.connectResolver = options;
         this.setConnecting();
-
-        await this.transport.connect();
+        try {
+            await this.transport.connect();
+        } catch (e) {
+            this.mqttDebug(`Transport connect error ("${this.transport.constructor.name}")`, e.message);
+            if (this.shouldReconnect()) {
+                return this.setDisconnected(e);
+            }
+            await this.setDisconnected(e);
+            throw e;
+        }
 
         this.createPipeline();
 
         return this.registerClient(await this.resolveConnectOptions());
     }
 
+    public async connect(options?: Resolvable<RegisterClientOptions>) {
+        try {
+            return await this._connect(options);
+        } catch (e) {
+            this.mqttDebug(`Connection error`, e.message);
+            this.emitError(e);
+            return e;
```

Comment:
Wouldn't this return a potential error in `transport.connect()`?

Reply:
Yes, it will **return** error thrown by `transport.connect()`, but will not throw it. It's by design.

Reply:
Then it's probably best to specify the return type explicitly.

Reply:
I'm generally an opponent of the `any` type, I am always in favor of explicitly specifying types. But previous `connect()` method signature is `Promise<any>`, so i don't know the actual return type of this method, i am still not an expert in `MQTTS`. Specifying `Promise<any | Error>`  makes no sense since it is the same as `Promise<any>`
If you could help me to specify `_connect` return type - i will be glad to specify `connect()` return type

Reply:
It's "bubbling" from `getConnectFlow` which specifies `unknown` as the success value. This could be changed to `ConnectResponsePacket` (and thus subsequent uses).